main.go

```diff
@@ -918,11 +921,16 @@ func writePidFile(loadedConfig *config.GatewayConfig, logger log.Logger) {
 		if err := ioutil.WriteFile(*loadedConfig.PidFilename, []byte(strconv.FormatInt(int64(pid), 10)), os.FileMode(0644)); err != nil {
 			logger.Log(log.Err, err, logkey.Filename, *loadedConfig.PidFilename, "cannot store pid in pid file")
 		}
-		// clean up the pid file
-		// also why are we removing the pid file?
+	} else {
```

Comment:
Does this actually happen given that the configuration has a default value?

Reply:
Nope, I removed the check for a nil PidFilename